protocol/x/accountplus/keeper/authenticators.go

```diff
@@ -106,3 +112,138 @@ func (k Keeper) SetNextAuthenticatorId(ctx sdk.Context, authenticatorId uint64)
 	store := ctx.KVStore(k.storeKey)
 	store.Set([]byte(types.AuthenticatorIdKeyPrefix), b)
 }
+
+// GetAuthenticatorExtension unpacks the extension for the transaction, this is used with transactions specify
+// an authenticator to use
+func (k Keeper) GetAuthenticatorExtension(exts []*codectypes.Any) types.AuthenticatorTxOptions {
+	for _, ext := range exts {
+		var authExtension types.AuthenticatorTxOptions
+		err := k.cdc.UnpackAny(ext, &authExtension)
+		if err == nil {
+			return authExtension
+		}
+	}
+	return nil
+}
+
+// GetSelectedAuthenticatorData gets all authenticators from an account
```

Comment:
Comment seems to deviate from behavior. Comment suggests all authenticators from a account; implementation gets a specific authenticator for an account

Reply:
Updated. Straight up copied from Osmosis w/o looking 😓